src/vs/workbench/contrib/inlineChat/browser/inlineChatController.ts

```diff
@@ -575,6 +584,19 @@ export class InlineChatController1 implements IEditorContribution {
 			barrier.open();
 		}));
 
+		if (options.attachments) {
+			await Promise.all(options.attachments.map(async attachment => {
+				if (/\.(png|jpe?g|gif|webp)$/i.test(attachment.path)) {
+					const context = await this.createImageAttachment(attachment);
+					if (context) {
+						this._ui.value.widget.chatWidget.attachmentModel.addContext(context);
+					}
+				} else {
+					this._ui.value.widget.chatWidget.attachmentModel.addFile(attachment);
```

Comment:
Just looked into the attachmentModel for another reason and found out that `addFile` already [handles images](https://github.com/microsoft/vscode/blob/46c465372682ae3a01bbe56c24a1b5acba01c020/src/vs/workbench/contrib/chat/browser/chatAttachmentModel.ts#L76)

So, the lines above don't seem needed

Reply:
yep, had to do some refactoring in `addFile` since the `isImageVariable` wasn't specific enough and only assumed it was coming from file URIs, but working nicely now